File: R/R/modify.R
#' Intelligently scan a data set and extract subject-level information
#'
#' @param df Data frame.
#' @param colnames Optional column-name mapping.
#' @return Subject-level data information.
#' @export
info_data <- function(df, colnames = NULL) {
    .help_info_data(df, colnames)
}

#' Modify and flatten model parameters
#'
#' @param user_params Optional user parameter specification.
#' @return Standardized and flattened model parameters.
#' @export
modify_params <- function(user_params = NULL) {
    .help_modify_params(user_params)
}

#' Modify and align prior distributions
#'
#' @param user_priors User prior specification.
#' @param std_params Optional standardized model parameters.
#' @return Standardized prior information.
#' @export
modify_priors <- function(user_priors, std_params = NULL) {
    .help_modify_priors(user_priors, std_params)
}

#' Evaluate the SDT model CDFs
#'
#' @param params Standardized model parameters.
#' @return Noise and signal CDF values.
#' @export
model_sdt <- function(params) {
    .core_model_sdt(params)
}

#' Evaluate the BCH model CDFs
#'
#' @param params Standardized model parameters.
#' @return Noise and signal CDF values and criteria.
#' @export
model_bch <- function(params) {
    .core_model_bch(params)
}

#' Evaluate the Normal meta-noise model probabilities
#'
#' @param params Standardized model parameters.
#' @return Model response-confidence probabilities.
#' @export
model_normal <- function(params) {
    .core_model_normal(params)
}

#' Evaluate the Log-normal meta-noise model probabilities
#'
#' @param params Standardized model parameters.
#' @return Model response-confidence probabilities.
#' @export
model_lognormal <- function(params) {
    .core_model_lognormal(params)
}

#' Evaluate the Signal Decay model probabilities
#'
#' @param params Standardized model parameters.
#' @return Model response-confidence probabilities.
#' @export
model_decay <- function(params) {
    .core_model_decay(params)
}

File: Python/metaSDT/_modify.py
"""Model helper wrappers for the Python frontend."""

from . import (
    _help_info_data,
    _help_modify_params,
    _help_modify_priors,
    _model_bch,
    _model_decay,
    _model_lognormal,
    _model_normal,
    _model_sdt,
)
from ._convert import to_backend_df_dict


def modify_params(user_params=None):
    return _help_modify_params.modify_params(user_params)


def modify_priors(user_priors, std_params=None):
    return _help_modify_priors.modify_priors(user_priors, std_params)


def model_sdt(std_params):
    return _model_sdt.model_sdt(std_params)


def model_bch(std_params):
    return _model_bch.model_bch(std_params)


def model_normal(std_params):
    return _model_normal.model_normal(std_params)


def model_lognormal(std_params):
    return _model_lognormal.model_lognormal(std_params)


def model_decay(std_params):
    return _model_decay.model_decay(std_params)


def info_data(df, colnames=None):
    if colnames is None:
        colnames = {}
    return _help_info_data.info_data(
        df=to_backend_df_dict(df),
        colnames=colnames,
    )
